tools/mpremote/mpremote/transport_serial.py

```diff
@@ -97,27 +101,33 @@ def close(self):
     def read_until(self, min_num_bytes, ending, timeout=10, data_consumer=None):
         # if data_consumer is used then data is not accumulated and the ending must be 1 byte long
         assert data_consumer is None or len(ending) == 1
+        assert isinstance(timeout, (type(None), int, float))
 
-        data = self.serial.read(min_num_bytes)
-        if data_consumer:
-            data_consumer(data)
-        timeout_count = 0
+        data = b""
+        begin_s = time.monotonic()
         while True:
+            if self.serial.inWaiting() <= 0:
+                # Wait for input
+                time.sleep(0.01)
+                continue
```

Comment:
Are you sure this won't wait forever if no characters come in at all at the beginning?